src/hotspot/share/gc/g1/g1CollectedHeap.cpp

```diff
@@ -980,7 +980,11 @@ class PostCompactionPrinterClosure: public HeapRegionClosure {
     : _hr_printer(hr_printer) { }
 };
 
-void G1CollectedHeap::print_hrm_post_compaction() {
+void G1CollectedHeap::print_heap_after_full_collection() {
```

Comment:
It's a bit odd to see `*_full_collection` as an API of `G1CollectedHeap`; I would have expected full-collection-only APIs in g1 full gc files.

Reply:
There are quite a few "after_full_collection" methods in `G1CollectedHeap`, everything below the
```
  // Internal helpers used during full GC to split it up to
  // increase readability.
```
comment. Ultimately there will always be young/full gc specific in `G1CollectedHeap` to avoid exposing too many details to the respective collector algorithm class. I do not know whether the current division is good, but I can and will revisit this at some point later.

Fwiw, the next PR will also add a few methods for young gc that are somewhat similar. Maybe we can group them in special per gc type `G1CollectedHeap` accessor classes?